go/vt/vtgate/evalengine/fn_time.go

```diff
@@ -1249,6 +1254,53 @@ func (call *builtinLastDay) compile(c *compiler) (ctype, error) {
 	return ctype{Type: sqltypes.Date, Flag: arg.Flag | flagNullable}, nil
 }
 
+func fromDays(loc *time.Location, d int64) *datetime.Date {
+	// 3652424 days corresponds to maximum date i.e. 9999-12-31
+	// mysql returns 0000-00-00 for days below 366
+	if d > 3652424 || d < 366 {
+		return nil
+	}
+
+	t := time.Date(1, time.January, 1, 0, 0, 0, 0, loc).AddDate(0, 0, int(d-366))
+	dt := datetime.NewDateFromStd(t)
+	return &dt
+}
+
+func (b *builtinFromDays) eval(env *ExpressionEnv) (eval, error) {
+	d, err := b.arg1(env)
+	if err != nil {
+		return nil, err
+	}
+	if d == nil {
+		return nil, nil
+	}
+	days := evalToInt64(d).i
+	dt := fromDays(env.currentTimezone(), days)
+
+	if dt == nil {
+		return nil, nil
+	}
+	return newEvalDate(*dt, env.sqlmode.AllowZeroDate()), nil
```

Comment:
Even if zero date mode is not allowed, MySQL here still returns zero dates so I don't think we should use that flag here, but always set `true` as the second argument (and we don't need a pointer as mentioned above).

Reply:
got it. done.